provider/bidengine/pricing.go

```diff
@@ -69,81 +70,70 @@ func (fp scalePricing) calculatePrice(ctx context.Context, gspec *dtypes.GroupSp
+	cpuTotal := decimal.NewFromInt(0)
+	memoryTotal := decimal.NewFromInt(0)
+	storageTotal := decimal.NewFromInt(0)
+	endpointTotal := decimal.NewFromInt(0)
 
 	// iterate over everything & sum it up
 	for _, group := range gspec.Resources {
+		groupCount := decimal.NewFromInt(int64(group.Count)) // Expand uint32 to int64
 
-		groupCount := big.NewInt(0)
-		groupCount.SetUint64(uint64(group.Count)) // Expand uint32 to uint64
-		groupCountRat := big.NewRat(0, 1)
-		groupCountRat.SetUint64(uint64(group.Count))
-
-		cpuQuantity := big.NewInt(0)
-		cpuQuantity.SetUint64(group.Resources.CPU.Units.Val.Uint64())
-		cpuQuantity.Mul(cpuQuantity, groupCount)
-		cpuTotal.Add(cpuTotal, cpuQuantity)
-
-		memoryQuantity := big.NewRat(0, 1)
-		memoryQuantity.SetUint64(group.Resources.Memory.Quantity.Value())
-		memoryQuantity.Mul(memoryQuantity, groupCountRat)
-		memoryTotal.Add(memoryTotal, memoryQuantity)
-
-		storageQuantity := big.NewRat(0, 1)
-		storageQuantity.SetUint64(group.Resources.Storage.Quantity.Val.Uint64())
-		storageQuantity.Mul(storageQuantity, groupCountRat)
-		storageTotal.Add(storageTotal, storageQuantity)
-
-		endpointQuantity := big.NewInt(0)
-		endpointQuantity.SetUint64(uint64(len(group.Resources.Endpoints)))
-		endpointTotal.Add(endpointTotal, endpointQuantity)
-	}
+		cpuQuantity := decimal.NewFromBigInt(group.Resources.CPU.Units.Val.BigInt(), 0)
+		cpuQuantity = cpuQuantity.Mul(groupCount)
+		cpuTotal = cpuTotal.Add(cpuQuantity)
+
+		memoryQuantity := decimal.NewFromBigInt(group.Resources.Memory.Quantity.Val.BigInt(), 0)
+		memoryQuantity = memoryQuantity.Mul(groupCount)
+		memoryTotal = memoryTotal.Add(memoryQuantity)
 
-	scale := big.NewInt(0)
-	scale.SetUint64(fp.cpuScale)
-	cpuTotal.Mul(cpuTotal, scale)
+		storageQuantity := decimal.NewFromBigInt(group.Resources.Storage.Quantity.Val.BigInt(), 0)
+		storageQuantity = storageQuantity.Mul(groupCount)
+		storageTotal = storageTotal.Add(storageQuantity)
+
+		endpointQuantity := decimal.NewFromInt(int64(len(group.Resources.Endpoints)))
+		endpointTotal = endpointTotal.Add(endpointQuantity)
+	}
 
-	mebibytes := big.NewRat(unit.Mi, 1)
-	memoryTotal.Quo(memoryTotal, mebibytes)
+	cpuTotal = cpuTotal.Mul(fp.cpuScale)
 
-	scaleRat := big.NewRat(0, 1)
-	scaleRat.SetUint64(fp.memoryScale)
-	memoryTotal.Mul(memoryTotal, scaleRat)
+	mebibytes := decimal.NewFromInt(unit.Mi)
 
-	storageTotal.Quo(storageTotal, mebibytes)
-	scaleRat.SetUint64(fp.storageScale)
-	storageTotal.Mul(storageTotal, scaleRat)
+	memoryTotal = memoryTotal.Div(mebibytes)
+	memoryTotal = memoryTotal.Mul(fp.memoryScale)
 
-	scale.SetUint64(fp.endpointScale)
-	endpointTotal.Mul(endpointTotal, scale)
+	storageTotal = storageTotal.Div(mebibytes)
+	storageTotal = storageTotal.Mul(fp.storageScale)
 
-	memoryTotalInt := ceilBigRatToBigInt(memoryTotal)
-	storageTotalInt := ceilBigRatToBigInt(storageTotal)
+	endpointTotal = endpointTotal.Mul(fp.endpointScale)
 
+	maxAllowedValue := decimal.NewFromInt(0x7fffffffffffffff)
```

Comment:
I think that there's a `math.MaxInt64` or something.

Reply:
Fixed.